### PR TITLE
Fix/pdf upload

### PR DESCRIPTION
**Pull request by karlknowit**
## Explain the changes you’ve made
Make it possible to upload PDF together with other files like images.

Also did smaller refactoring.

## Explain why these changes are made
A user needs to be able to upload PDF when answering the form, hence adding the ability to do so.

## Explain your solution
Added new properties to the pdfUpload component to make it more alike the imageUploader component. This to make it easier in the future to refactor the file selector components to only one. 
Also by doing so we can remove the uploading logic from both pdfUploader and imageUploader components to fileUploader components instead.

## How to test

1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios` or `yarn android`
3. Make sure you have a form that supports both imageUploader and PdfUploader
4. Add files and images (or only one of them)
5. When uploading the files, check the number of files uploaded in the uploading info modal

## Tested environments

- [] Storybook on a iOS device/simulator.
- [x] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots
simulator_screenshot_C76E233A-4538-4083-9AE0-3F3246DF764E

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1pe0y8x Komponent för att ladda upp PDF-filer i appen](https://app.clickup.com/t/1pe0y8x)

**Review by MasterKenth — CHANGES_REQUESTED** [review]
I selected 2 images with label, 2 images without, 2 pdfs with label, 2 pdfs without. During upload it failed:

image

File: source/components/molecules/PdfDisplay/PdfDisplay.tsx [hunk 1]
```diff
@@ -22,59 +22,81 @@ export interface Pdf extends DocumentPickerResponse {
...
 }
 
 const PdfDisplay: React.FC<Props> = ({ pdfs, answers, onChange }) => {
-  const [horizontalScrollPercentage, setHorizontalScrollPercentage] = useState(0);
+  const [horizontalScrollPercentage, setHorizontalScrollPercentage] =
+    useState(0);
 
-  const deletePdfFromCloudStorage = async (pdf: Pdf) => {
-    console.log('Placeholder: not implemented yet in API, want to delete pdf ', pdf.name, pdf.uri);
+  const deletePdfFromCloudStorage = async (pdf: UploadedPdf) => {
```
Comment (MasterKenth) [inline source/components/molecules/PdfDisplay/PdfDisplay.tsx:50]: `deletePdfFromCloudStorage` can be removed.

File: source/components/molecules/PdfUploader/PdfUploader.tsx [hunk 2]
```diff
@@ -23,78 +28,96 @@ const ButtonContainer = styled.View`
...
 
-      const pdf: Pdf = { questionId: id, ...fileInfo };
-      const originalLength = pdfs.length;
-      const updatedPdfs = addPdf(pdf);
-      uploadPdf(pdf, originalLength, updatedPdfs);
+      onChange(filesWithQuestionId);
     } catch (error) {
-      if (!DocumentPicker.isCancel(error))
-        console.error('Error while adding pdf from library:', error);
+      if (!DocumentPicker.isCancel(error as Error)) {
```
[the remark this answers is not in the document]
Comment (karlknowit) [reply source/components/molecules/PdfUploader/PdfUploader.tsx:98]: You tell me 😉

**Review by MasterKenth — APPROVED** [review]

**Pull request merged**